src/Testing/PHPUnit/AbstractRectorTestCase.php

```diff
@@ -227,6 +232,12 @@ private function doTestFileMatchesExpectedContent(
             // if not exact match, check the regex version (useful for generated hashes/uuids in the code)
             $this->assertStringMatchesFormat($expectedFileContents, $changedContents, $failureMessage);
         }
+
+        if ($inputFileContents === $expectedFileContents && $numAppliedRectorClasses > 0) {
```

Comment:
If we expect that the code should not change but there has been any Rector rule applied then we throw an error to let the user know that this is incorrect